sympy/core/add.py

```diff
@@ -140,14 +140,14 @@ def flatten(cls, seq):
                     # we know for sure the result will be nan
                     return [S.NaN], [], None
                 if coeff.is_Number or isinstance(coeff, AccumBounds):
-                    coeff = coeff + o
+                    coeff += o
                     if coeff is S.NaN and not extra:
                         # we know for sure the result will be nan
                         return [S.NaN], [], None
                 continue
 
             elif isinstance(o, AccumBounds):
-                coeff = o.__add__(coeff)
+                coeff += o
```

Comment:
I think that this was coded this way for a reason. If something breaks, this should probably be left as is.

Reply:
It works

Reply:
Yes, it seems to work because the `_op_priority` of `AccumBounds` is higher than that of ordinary expressions. But it will run through a lot of extra code before coming finally to `o.__add__(coeff)`. So I think that it was coded this way for efficiency.

Reply:
Doesn't the same apply to the [other occurrence](https://github.com/sympy/sympy/pull/18478/files#diff-9534cbdadda1fd255f8b3683e7b9124bR142-R143)?
```python
coeff = coeff + o if o.is_Number else o.__add__(coeff)
```
or is it the `+=` operator that is the culprit?

Reply:
I think there is not much difference in using either of them as:
Using `coeff = coeff + o if o.is_Number else o.__add__(coeff)`, I get,
```
In [1]: %timeit Add(*[AccumBounds(-1,1), oo])
Out[1]: 56.9 µs ± 325 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```
And Using `coeff+=o`, 
```
In [1]: %timeit Add(*[AccumBounds(-1,1), oo])
Out[1]: 57.3 µs ± 391 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
> Doesn't the same apply to the other occurrence?

On line 143, we know that `o` is `Number`, so if `coeff` is `AccumBounds`, `coeff += o` would call `AccumBounds.__add__`, which is what we want. On line 150, the roles have been interchanged. Then `coeff += o` will also finally lead to `AccumBounds.__add__` but only via `Number.__add__` etc. I know it is not much of an optimization but that is probably the reason why it was coded that way.

Reply:
Thanks for the clarification, @jksuom .